### ISSUE TITLE
Testing useWorker

### ISSUE THREAD

**Issue by zant** [alewin/useWorker#41]
Since they are plans to add new features (#36) I started to work in adding some tests for the hook. I initially setup [Jest](https://jestjs.io/) with [react-hooks-testing-library](https://github.com/testing-library/react-hooks-testing-library). But found that Jest (which uses [jsdom](https://github.com/jsdom/jsdom)) does not have an implementation for [URL.createObjectURL](https://github.com/jsdom/jsdom/issues/1721) nor [Web Workers](https://github.com/facebook/jest/issues/3449). 

So I had to dig a bit on the internet 🌈to find potential workarounds, and it seems that they're two approaches to test workers (afaik) that we can use.

### Options

- Test with [karma](https://karma-runner.github.io/4.0/index.html) on [headless chrome](https://developers.google.com/web/updates/2017/04/headless-chrome), this approach is used by [comlink](https://github.com/GoogleChromeLabs/comlink/tree/master/tests) which is a wrapper for Web Workers. I did try this and it worked nicely, the one downside I see is the tooling (needs a preprocessor like webpack with babel to bundle everything for the test, as it's going to be tested in a browser) but at the end they're just devDependencies so it will not affect bundle size. 
- Mock the Web Worker with something like [jsdom-worker](https://github.com/developit/jsdom-worker) (although looks like is not actively maintained). This package also comes with a `URL.createObjectURL` mock :).

Imo, I prefer karma since it will provide reliable behavior for the tests. What do you guys think? 😄

For reference, this is the test I tried with Karma:

```js
import React from "react";
import { useWorker } from "../dist/index";
import { renderHook } from "@testing-library/react-hooks";

it("Basic Test", async () => {
  const sum = (a, b) => a + b;
  const { result } = renderHook(() => useWorker(sum));
  const [sumWorker] = result.current;
  const res = await sumWorker(1, 2);
  assert.equal(res, 3);
});
```

**Comment by alewin** [comment]
Perfect analysis @gonzachr thanks! 👏 I like the first option (karma approach), since it's based on a headless browser, moreover `react-hooks-testing-library` is really simple to use.

**Comment by gonzachr** [comment]
Awesome @alewin! I'll open a PR later today 👍

**Comment by gonzachr** [comment]
Admittedly at first I just tried the first preprocessing option I found. Today I will make a few experiments with rollup as I found another [karma preprocessor](https://github.com/jlmakes/karma-rollup-preprocessor) compatible with it. I heard good things about rollup specially for libraries so I think is a nice option, becase it really feels a bit bloated to add Webpack just for testing. Specially having such a compact bundler for the project itself 😅
